### ISSUE TITLE
Approximation en dur des Cotisations Famille et MMID des indépendants (artisans-commercants) 

### ISSUE THREAD

**Issue by Sasha-Laniece** [openfisca/openfisca-france#1775]
Hello hello !
Je suis le fan numéro un d'OpenFisca, mais je viens de rencontrer un problème.

### Qu'ai-je fait ?
En observant les cotisations Famille et MMID des indépendants (artisans-commercants) [dans la PR](https://github.com/openfisca/openfisca-france/pull/1761) on observe quelques problèmes:

[FAMILLE] @Sasha-Laniece 
Pour cette partie, on a des choses à modifier pour etre à jour:
- [  ] le taux de cotisation a évolué en 2015 pour un bareme plus progressif : https://www.urssaf.fr/portail/home/independant/mes-cotisations/quelles-cotisations/la-cotisation-allocations-famili/base-et-taux-de-la-cotisation-da.html
- [  ] le bareme d'avant (1974-1983) n'est pas codé. Choix fait lors de l'harmonisaiton: je laisse l'ancien bareme comme il est et j'ajoute tous les taux commes parametres simples (dans famille.famille_ind), mais il faudra ensuite en faire un unique bareme

[FAMILLE et MMID] @clallemand 
J'ai l'impression que le plus gros problème à noter et c'est que les paramètres ne sont pas appelées dans les formules. Les paramètres y sont écrits "en dur".

En effet, on a des taux et des ratios en dur dans le code , comme ici (famille):
 https://github.com/openfisca/openfisca-france/blob/f28f66938fbfe6580a44ff63f67514064be8d911/openfisca_france/model/prelevements_obligatoires/prelevements_sociaux/cotisations_sociales/travail_non_salarie.py#L263-L307

ou là (MMID) :
https://github.com/openfisca/openfisca-france/blob/f28f66938fbfe6580a44ff63f67514064be8d911/openfisca_france/model/prelevements_obligatoires/prelevements_sociaux/cotisations_sociales/travail_non_salarie.py#L141-L185 

[Commentaire de @pzuldp ](https://github.com/openfisca/openfisca-france/pull/1761#discussion_r778069098)
je me demande comment faire pour inclure l'intégralité du barème de cotisations. En fait, à partir de 2017 (2018) en dessous de 70% (110%) du PASS, on n'a pas de barèmes, alors que (bien sûr) le taux n'est pas nul. La formule est mal adaptée au format des paramètres, car la loi indique la formule de taux qui est croissant avec l'assiette (mais il n'y a pas de "barèmes", c'est à dire de taux marginal explicite). C'est la raison pour laquelle les paramètres ne sont pas appelés dans les formules comme le soulevait @clallemand.

- [ ] Contributeur·e : je contribue à OpenFisca France.
- [ ] Développeur·e : je crée des outils qui utilisent OpenFisca France.
- [ ] Économiste : je réalise des simulations avec des données.
- [ ] Mainteneur·e : j'intègre les contributions à OpenFisca France.
- [ ] Autre : _(ajoutez une description du contexte dans lequel vous utilisez OpenFisca)_.

**Comment by pzuldp** [comment]
Oui, la formule donnée pour la cotis mmid dans la loi ([par exemple pour 2018](https://www.legifrance.gouv.fr/jorf/id/JORFARTI000036342512)) est un peu différente de d'habitude, on a un taux de cotisation dont la valeur est donnée par une formule qui dépend non-linéairement du niveau de l'assiette. Mais cette formule ne passe pas par l'application d'un barème de taux marginaux. 
- Une manière de coder ce barème serait de récupérer tous les paramètres de la formule, ce qui a pour avantage de "coller" au texte de loi, et le désavantage de coder un taux de manière assez inhabituelle.
- Une autre manière serait de reverse-engineerer le barème qui correspondrait à la formule. Ce qui a les avantages et désavantages symétriques à supra (et un désavantage supplémentaire qu'il faut faire le calcul :) ). 
Je vote pour la première manière :)

**Comment by pzuldp** [comment]
par contre je ne suis pas sûr de comprendre en quoi les formules actuelles sont des approximations @Sasha-Laniece ? (en fait en relisant l'issue c'est peutêtre surtout sur "famille" que porte ce terme ?)
